src/expand.ts

```diff
@@ -103,17 +103,21 @@ export async function expandString<T>(tmpl: string | T, opts: ExpansionOptions):
     const MAX_RECURSION = 10;
     let result = tmpl;
     let didReplacement = false;
+    let circularReference: string | undefined;
 
     let i = 0;
     do {
         // TODO: consider a full circular reference check?
         const expansion = await expandStringHelper(result, opts);
         result = expansion.result;
         didReplacement = expansion.didReplacement;
+        circularReference = expansion.circularReference;
         i++;
-    } while (i < MAX_RECURSION && opts.recursive && didReplacement);
+    } while (i < MAX_RECURSION && opts.recursive && didReplacement && !circularReference);
 
-    if (i === MAX_RECURSION) {
+    if (circularReference) {
+        log.warning(localize('circular.variable.reference', 'Circular variable reference found: {0}', circularReference));
```

Comment:
This gives the string with the reference, like "${PATH}", but not the variable name that went with it.  Can we save `full` and `varname` down by line 183?